pkg/urbit/vere/io/ames.c

```diff
@@ -1149,7 +1168,16 @@ _ames_hear(u3_ames* sam_u,
   else {
     u3_noun msg = u3i_bytes(len_w, hun_y);
     c3_free(hun_y);
-    _ames_put_packet(sam_u, msg, *lan_u);
+#ifdef AMES_SKIP
+    if (_ames_skip(&bod_u) == c3y ) {
+      u3z(msg);
+    }
+    else {
+#endif
+      _ames_put_packet(sam_u, msg, *lan_u);
+#ifdef AMES_SKIP
+    }
+#endif
```

Comment:
I'd be tempted to refactor this block so that we only allocate `msg` if we're going to use it. But it's hard to imagine that actually mattering very much ...

Reply:
Yeah, arguably it should all move up to right after `bod_u` is defined but with early return you want to make it obvious you're not leaking memory, so I did it at the last possible second.